modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/TransactionStateResolver.java

```diff
@@ -213,18 +215,22 @@ private void resolveTxStateFromTxCoordinator(
         updateLocalTxMapAfterDistributedStateResolved(txId, txMetaFuture);
 
         InternalClusterNode coordinator = coordinatorId == null ? null : clusterNodeResolver.getById(coordinatorId);
-
         if (coordinator == null) {
             // This means the coordinator node have either left the cluster or restarted.
             markAbandoned(txId);
 
             resolveTxStateFromCommitPartition(txId, commitGrpId, txMetaFuture);
         } else {
-            txMessageSender.resolveTxStateFromCoordinator(coordinator.name(), txId, timestamp)
+            txMessageSender.resolveTxStateFromCoordinator(coordinator, txId, timestamp)
                     .whenComplete((response, e) -> {
-                        if (e == null) {
-                            txMetaFuture.complete(asTransactionMeta(response.txStateMeta()));
+                        if (e == null && response.txStateMeta() != null) {
+                            TransactionMetaMessage transactionMetaMessage = Objects.requireNonNull(response.txStateMeta(),
+                                    "Transaction state meta must not be null after check");
```

Comment:
Very minor: full stop is missing. As far as I remember according to out code guide lines we should add '.' in the end of the sentence.